src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -5614,10 +5649,35 @@ public void explore(ExternalTool externalTool) {
             apiToken = new ApiToken();
             apiToken.setTokenString(privUrl.getToken());
         }
+        return apiToken;
+    }
+    
+    public void explore(ExternalTool externalTool) {
+        ApiToken apiToken = getApiTokenForTool();
         ExternalToolHandler externalToolHandler = new ExternalToolHandler(externalTool, dataset, apiToken, session.getLocaleCode());
         String toolUrl = externalToolHandler.getToolUrlWithQueryParams();
         logger.fine("Exploring with " + toolUrl);
         PrimeFaces.current().executeScript("window.open('"+toolUrl + "', target='_blank');");
     }
+    
```

Comment:
other than the original method being named 'explore', is there a reason multiple methods are needed? Unless I missed something, they all do the same thing. Perhaps explore() could be 'launchTool' or similar?